### DISCUSSION TITLE
Lots of duplicate implementation warning logs at launch

### DISCUSSION THREAD

**Discussion by gutley** [mozilla-mobile/firefox-ios#13730] [General]
When running from the 111.0 branch or main there are a lot of these warnings at launch:

objc[52797]: Class _TtC6Common21BrowserKitInformation is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcf9f00) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7878). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common12AppContainer is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa020) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7998). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common19DefaultCrashManager is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa180) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7af8). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common13DefaultLogger is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa350) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7cc8). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common24DefaultLoggerFileManager is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa480) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7df8). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common13DefaultSentry is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa590) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7f08). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common7AppInfo is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa658) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc7fd0). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common5Bytes is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa6e8) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc8060). One of the two will be used. Which one is undefined.
objc[52797]: Class _TtC6Common10DeviceInfo is implemented in both <Snipped>/Fennec-iphonesimulator/PackageFrameworks/Common_17A2E8D93C1D2F_PackageProduct.framework/Common_17A2E8D93C1D2F_PackageProduct (0x10dcfa780) and /Users/mgough/Library/Developer/CoreSimulator/Devices/9BCBD00B-AE2C-489C-BCDF-101ABF545639/data/Containers/Bundle/Application/38301FC3-8AFD-4E66-915B-892FD774B13D/Client.app/Client (0x10bbc80f8). One of the two will be used. Which one is undefined.

@lmarceau Anything that can be done?

**Comment by lmarceau** [comment]
Hey @gutley, thanks for raising this. We'll have a look!
